tp/src/handler.rs

```diff
@@ -668,9 +700,15 @@ fn create_namespace_registry(
         }
     }
 
-    let mut namespace_registry = NamespaceRegistry::new();
-    namespace_registry.set_namespace(namespace.into());
-    namespace_registry.set_owners(RepeatedField::from_vec(payload.get_owners().to_vec()));
+    let namespace_registry = NamespaceRegistryBuilder::new()
+        .with_namespace(namespace.into())
+        .with_owners(payload.owners().to_vec())
+        .build()
+        .map_err(|_| {
+            ApplyError::InvalidTransaction(String::from(
+                "Cannot build namespace registry permission",
```

Comment:
Is this actually a permission or just a namespace registry?

Reply:
just registry.